client/network/src/transport.rs

```diff
@@ -145,5 +155,121 @@ pub fn build_transport(
 			.map_err(|err| io::Error::new(io::ErrorKind::Other, err))
 			.boxed();
 
-	(transport, sinks)
+	(transport, bandwidth)
+}
+
+/// A monitor for network bandwidth usage.
+pub struct BandwidthMonitor {
```

Comment:
Out of scope for this pull request, but I think worth mentioning:

While likely useful for the RPC `NetworkState` reporting, I don't think reporting a moving average to Prometheus is a good idea as it blurs the data itself. Instead I would suggest reporting the plain total bytes in / out to Prometheus and have Prometheus do the moving average on-demand if required.

Reply:
I agree, I just didn't want to change the `NetworkState` in this PR, as it is public API.

Reply:
I actually went ahead and changed `NetworkState` together with the metrics.